lib/src/test/groovy/com/cloudbees/groovy/cps/CpsTransformerTest.groovy

```diff
@@ -777,4 +777,18 @@ class CpsTransformerTest extends AbstractGroovyCpsTest {
         ''') == 'FOO';
     }
 
+    @Issue("JENKINS-38268")
+    @Test
+    void lexicalScope() {
+        assert evalCPS('''
+def a = [id: 'a', count: 0]
+def b = [id: 'b', count: 0]
+
+def toRun = [a, b].collect { thing -> return { thing.count = thing.count + 1 } }
```

Comment:
Is there a simpler variant that would demonstrate the problem without `[Cps]DefaultGroovyMethods`?